### PR TITLE
simplify controller wiring

### PR DESCRIPTION
**Pull request by deads2k**
This builds on @mfojtik's pull to simplify the wiring and reduce us to a single list.

@enj if you have time to review before dcut, you'll thank yourself

### CODE REVIEW

**Comment by deads2k** [comment]
[test]

File: pkg/cmd/server/start/start_master.go [hunk 1]
```diff
@@ -594,13 +594,58 @@ func startControllers(oc *origin.MasterConfig, kc *kubernetes.MasterConfig) erro
...
+			},
+		},
+		TemplateInformers:            oc.TemplateInformers,
+		DeprecatedOpenshiftInformers: oc.Informers,
+		Stop: utilwait.NeverStop,
+	}
+
+	// We need to start the serviceaccount-tokens controller first as it provides token
+	// generation for other controllers.
+	preStartControllers, err := oc.NewOpenShiftControllerPreStartInitializers()
```
Comment (enj) [inline pkg/cmd/server/start/start_master.go]: No error check?

**Comment by mfojtik** [comment]
@deads2k fixed @enj and rebased against Clayton informers in service account.

File: pkg/cmd/server/origin/controller.go [hunk 2]
```diff
@@ -69,21 +69,23 @@ func (c *MasterConfig) NewOpenShiftControllerPreStartInitializers() (map[string]
 		}
 		saTokens.ServiceServingCA = append(saTokens.ServiceServingCA, serviceServingCA...)
 	}
-	ret["serviceaccount-tokens"] = saTokens.RunController
+	// this matches the upstream name
```
Comment (mfojtik) [inline pkg/cmd/server/origin/controller.go]: can we rename the controller as well?

File: pkg/cmd/server/origin/controller.go [hunk 3]
```diff
@@ -69,21 +69,23 @@ func (c *MasterConfig) NewOpenShiftControllerPreStartInitializers() (map[string]
...
 
+	// TODO this overrides an upstream controller, so move this to where we initialize upstream controllers
 	serviceAccount := controller.ServiceAccountControllerOptions{
 		ManagedNames: c.Options.ServiceAccountConfig.ManagedNames,
 	}
 	ret["serviceaccount"] = serviceAccount.RunController
 
-	ret["serviceaccount-pull-secrets"] = controller.RunServiceAccountPullSecretsController
-	ret["origin-namespace"] = controller.RunOriginNamespaceController
+	ret["openshift.io/serviceaccount-pull-secrets"] = controller.RunServiceAccountPullSecretsController
```
Comment (mfojtik) [inline pkg/cmd/server/origin/controller.go]: openshift.io/controllers/ ? 

how about some generator with:

```
ret.AddOpenShiftController(controller.RunServiceAccount...)
ret.AddKubeController(...)

return ret.InitializersMap()
```

too much?

**Comment by mfojtik** [comment]
2 small nits,  LGTM after rebase and green test :-)

File: pkg/cmd/server/origin/controller.go (hunk 3 above)
Comment (deads2k) [inline pkg/cmd/server/origin/controller.go]: > openshift.io/controllers/ ?
> 
> how about some generator with:
> 
> ret.AddOpenShiftController(controller.RunServiceAccount...)
> ret.AddKubeController(...)
> 
> return ret.InitializersMap()
> too much?

I think we actually end up creating a method to create `NewKubeControllerInitializers` and then one method that does `NewAllControllerInitializers` or something.  That will help us prep for splitting apart.

File: pkg/cmd/server/origin/controller.go (hunk 2 above)
Comment (deads2k) [inline pkg/cmd/server/origin/controller.go]: > can we rename the controller as well?

doing.

**Comment by deads2k** [comment]
comments addressed. [merge]

**Comment by deads2k** [comment]
included in https://github.com/openshift/origin/issues/14317

**Pull request closed without merging**